src/passes/TypeSSA.cpp

```diff
@@ -170,28 +293,47 @@ struct TypeSSA : public Pass {
       return;
     }
 
-    // First, find all the struct/array.news.
+    struct Info {
+      News news;
+      ExactTypes exactTypes;
+    };
 
-    ModuleUtils::ParallelFunctionAnalysis<News> analysis(
-      *module, [&](Function* func, News& news) {
+    // First, find all the struct/array.news.
```

Comment:
```suggestion
    // First, analyze all the functions.
```
(as this does more than before)